[Kubernetes] Remove extra base fields for state datastreams

constanca-m opened the pull request:
## Proposed commit message

- WHAT: remove extra base fields for all `state_*` datastreams.
- WHY:  we have many fields for the data streams that never hold any value or are not related to it.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How this was tested

1. Created a cluster with the configuration `kind-config.yaml` present in this repo.
2. Deployed all resources in `kubernetes/_dev/deploy/k8s` (these are also the resources being used in testing). Modified some to add annotations.
3. The Kubernetes integrations has all data streams enabled.



## Bugs and warnings

There are some things that don't seem to match expected behaviour.

The most significant one is this:

- `kubernetes.annotations.*` never holds value for any `state_*` datastream. I tried to add `metadata.annotations` to the objects, but none were reported:
  - Is this a bug or expected behavior?
  - I am leaving this group field in all of them, as I am unsure of the answer for the above.

Proof:

![Screenshot from 2023-11-03 11-09-12](https://github.com/elastic/integrations/assets/113898685/50044b1b-eaf2-485b-a7cb-52d265768496)



And if I check (for example) one PVC in the cluster, I can see annotations there:

![Screenshot from 2023-11-03 11-32-20](https://github.com/elastic/integrations/assets/113898685/ba8b0f3f-5687-41a3-878b-c10973835c6b)

Same for a job:

![Screenshot from 2023-11-03 11-40-16](https://github.com/elastic/integrations/assets/113898685/08ded0b9-ffb4-4f36-ad1d-accb2d6ffd15)

And so on.


Other unexpected behaviours:
- Resource quota does not seem to hold `kubernetes.labels.*` even when they were added to the manifest file.
- `state_pod` holds data for `node.labels.*` but not for `node.annotations.*` (same problem as `kubernetes.annotations.*`). This field was removed.
- The same is happening for `namespace_labels.*` and `namespace_annotations.*`. I added the warning separately for these data streams.


## Changes

**NOTE**: Take into account the group field `kubernetes.annotations.*` was not removed from any data stream even if no results were reported. This seems like a bug that needs to be fixed.


For each data stream, the following fields were removed:

<details>
<summary>
state_container
</summary>

  - `kubernetes.node.annotations.*`
  - `kubernetes.namespace_annotations.*`
  - `kubernetes.selectors.*`
  - `kubernetes.container.image`

</details>

- **Warning**: `kubernetes.deployment.name` had no values, even though there were deployments in the cluster... Maybe there is a bug in the code, as it does not seem to be expected. I did not remove this field. Same for `kubernetes.cronjob.name`.


<details>
<summary>
state_cronjob
</summary>

  - `kubernetes.pod.name`
  - `kubernetes.pod.uid`
  - `kubernetes.pod.ip`
  - `kubernetes.node.name`
  - `kubernetes.node.hostname`
  - `kubernetes.selectors.*`
  - `kubernetes.replicaset.name`
  - `kubernetes.deployment.name`
  - `kubernetes.statefulset.name`
  - `kubernetes.container.name`
  - `kubernetes.container.image`

</details>


<details>
<summary>
state_daemonset
</summary>

  - `kubernetes.pod.name`
  - `kubernetes.pod.uid`
  - `kubernetes.pod.ip`
  - `kubernetes.node.name`
  - `kubernetes.node.hostname`
  - `kubernetes.selectors.*`
  - `kubernetes.replicaset.name`
  - `kubernetes.deployment.name`
  - `kubernetes.statefulset.name`
  - `kubernetes.container.name`
  - `kubernetes.container.image`

</details>


<details>
<summary>
state_deployment
</summary>

  - `kubernetes.pod.name`
  - `kubernetes.pod.uid`
  - `kubernetes.pod.ip`
  - `kubernetes.node.name`
  - `kubernetes.node.hostname`
  - `kubernetes.selectors.*`
  - `kubernetes.replicaset.name`
  - `kubernetes.statefulset.name`
  - `kubernetes.container.name`
  - `kubernetes.container.image`

</details>


<details>
<summary>
state_job
</summary>

  - `kubernetes.pod.name`
  - `kubernetes.pod.uid`
  - `kubernetes.pod.ip`
  - `kubernetes.node.name`
  - `kubernetes.node.hostname`
  - `kubernetes.selectors.*`
  - `kubernetes.replicaset.name`
  - `kubernetes.deployment.name`
  - `kubernetes.statefulset.name`
  - `kubernetes.container.name`
  - `kubernetes.container.image`
</details>


<details>
<summary>
state_namespace
</summary>
Newly introduced, nothing to delete.
</details>

<details>
<summary>
state_node
</summary>

  - `kubernetes.pod.name`
  - `kubernetes.pod.uid`
  - `kubernetes.pod.ip`
  - `kubernetes.namespace`
  - `kubernetes.node.hostname`
  - `kubernetes.selectors.*`
  - `kubernetes.replicaset.name`
  - `kubernetes.deployment.name`
  - `kubernetes.statefulset.name`
  - `kubernetes.container.name`
  - `kubernetes.container.image`
</details>



<details>
<summary>
state_persistentvolume
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.namespace`
- `kubernetes.node.name`
- `kubernetes.node.hostname`
- `kubernetes.selectors.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>


<details>
<summary>
state_persistentvolumeclaim
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.namespace`
- `kubernetes.node.name`
- `kubernetes.selectors.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>


<details>
<summary>
state_pod
</summary>

- `kubernetes.node.annotations.*`
- `kubernetes.namespace_annotations.*`
- `kubernetes.selectors.*`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>

- **Warning**: `kubernetes.deployment.name` had no values, even though there were deployments in the cluster... Maybe there is a bug in the code, as it does not seem to be expected. I did not remove this field. Same for `kubernetes.cronjob.name`.
- **Warning**: there is data for `node.labels.*` but not for `node.annotations.*` (same problem as `kubernetes.annotations.*`). This field was removed. 
- **Warning**: same case for `namespace_labels.*` and `namespace_annotations.*`. This field was also removed.

<details>
<summary>
state_replicaset
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.node.name`
- `kubernetes.node.hostname`
- `kubernetes.selectors.*`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>



<details>
<summary>
state_resourcequota
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.node.name`
- `kubernetes.selectors.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>



<details>
<summary>
state_service
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.namespace_annotations.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>

- **Warning**: `namespace_labels.*` and `namespace_annotations.*`: one has value, the other has not. Expected?... This field was also removed.

<details>
<summary>
state_statefulset
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.node.name`
- `kubernetes.node.hostname`
- `kubernetes.seletectors.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.seletectors.*`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>


<details>
<summary>
state_storageclass
</summary>

- `kubernetes.pod.name`
- `kubernetes.pod.uid`
- `kubernetes.pod.ip`
- `kubernetes.node.name`
- `kubernetes.node.hostname`
- `kubernetes.namespace`
- `kubernetes.seletectors.*`
- `kubernetes.replicaset.name`
- `kubernetes.deployment.name`
- `kubernetes.statefulset.name`
- `kubernetes.container.name`
- `kubernetes.container.image`
</details>


## Results

Expected result is that nothing will be broken and everything will keep running as before.

I built the package and update the policy. It was updated as expected:
![Screenshot from 2023-11-03 11-59-43](https://github.com/elastic/integrations/assets/113898685/833f25ec-ed0c-45e9-8d8e-1aa2055b5292)


I also checked every dashboard and all were still working as before (not including screenshots to not overwhelm this description).

elasticmachine commented:
## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_8393.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-11-03T11:02:46.688Z&rangeTo=2023-11-03T11:22:46.688Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=4a6aabb00ad7d87ce13c7e4a7bb86510&transactionId=5c796f15f12d0c61)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-11-03T11:12:46.688+0000


* Duration: 17 min 53 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 1  |
| Passed       | 72  |
| Skipped      | 0 |
| Total        | 73   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![1](https://img.shields.io/badge/1%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `static test: Verify sample_event.json – kubernetes.state_container`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 null 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 one or more errors found in document: [0] field "kubernetes.replicaset.name" is undefined
[1] field "kubernetes.deployment.name" is undefined
[2] field "kubernetes.container.name" is undefined
[3] field "kubernetes.pod.uid" is undefined
[4] field "kubernetes.pod.ip" is undefined
[5] field "kubernetes.pod.name" is undefined
[6] field "kubernetes.node.uid" is undefined
[7] field "kubernetes.node.hostname" is undefined
[8] field "kubernetes.node.name" is undefined
[9] field "kubernetes.node.labels.kubernetes_io/arch" is undefined
[10] field "kubernetes.node.labels.kubernetes_io/os" is undefined
[11] field "kubernetes.node.labels.beta_kubernetes_io/arch" is undefined
[12] field "kubernetes.node.labels.node_kubernetes_io/exclude-from-external-load-balancers" is undefined
[13] field "kubernetes.node.labels.node-role_kubernetes_io/master" is undefined
[14] field "kubernetes.node.labels.kubernetes_io/hostname" is undefined
[15] field "kubernetes.node.labels.node-role_kubernetes_io/control-plane" is undefined
[16] field "kubernetes.node.labels.beta_kubernetes_io/os" is undefined 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8393/detail/PR-8393/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Test integration: kubernetes`
<ul>
<li>Took 0 min 8 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-8393/runs/1/steps/1236/log/?start=0">here</a></li>
<li>Description: <code>                                eval "$(../../build/elastic-package stack shellinit)"                                ../../build/elastic-package test -v --report-format xUnit --report-output file --test-coverage                                </code></l1>
</ul>
  
  
  
  
##### `Google Storage Download`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-8393/runs/1/steps/1326/log/?start=0">here</a></li>
<li>Description: <code>null</code></l1>
</ul>
  
  </p>
  </details>



## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`1/1`) | :green_heart: 
Files | 100.0% (`1/1`) | :green_heart: 
Classes | 100.0% (`1/1`) | :green_heart: 
Methods | 100.0% (`56/56`) | :green_heart: 
Lines | 100.0% (`22/22`) | :green_heart: 
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

gizas commented:
>Warning: kubernetes.deployment.name had no values, even though there were deployments in the cluster... Maybe there is a bug in the code, as it does not seem to be expected. I did not remove this field. Same for kubernetes.cronjob.name.

Since elastic-agent-autodiscovery 0.6.4 (https://github.com/elastic/elastic-agent-autodiscover/commit/285f0bb651192115743c7764d60e16e5fe4a0ef9) we disable deployment and cronjob enrichemnt by default.

So you need to enable those specifically in the advance options to make the deployment and cronjob name appear.

> but not for node.annotations.* (same problem as kubernetes.annotations.*). 

The annotations are not include by default. See documentation here: https://www.elastic.co/guide/en/fleet/current/add_kubernetes_metadata-processor.html

So you need to add include_annotations under node and namespace fields

FYI there is also this [bug](https://github.com/elastic/elastic-agent/issues/3636) open that is related

gizas commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -89,56 +44,7 @@
       description: >
         Kubernetes namespace labels map
 
-    - name: namespace_annotations.*
```
We need those when add_resource_metadata.namespace.include_labels is defined

constanca-m replied:
I added this:

```
add_resource_metadata:
  namespace:
    include_labels: ["namespacelabel1"]
    include_annotations: ["namespaceannotation1"]
```

But still I cannot see it in Discover. I added the mapping anyway, maybe I am doing it wrong.

constanca-m replied:
I added it again. I also tested it, but I cannot see it :confused:  Maybe I am doing something wrong on this

constanca-m replied:
I see it now in one of my tests. Thanks for pointing it out, it exists indeed!

gizas commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -89,56 +44,7 @@
-      description: >
-        Kubernetes namespace annotations map
-
-    - name: selectors.*
```
Do We also need those when https://www.elastic.co/guide/en/fleet/current/kubernetes-provider.html resource type is service ?

constanca-m replied:
From my understanding, only `state_service` would have it, even the description seems to imply it. I will test it again

ChrsMark replied:
👍🏽 See the implementation PR: https://github.com/elastic/beats/pull/23730

gizas commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -89,56 +44,7 @@
       description: >
         Name of the CronJob to which the Pod belongs
-
-    - name: container.name
```
We need this is not it?

constanca-m replied:
I cannot find results for this:
![image](https://github.com/elastic/integrations/assets/113898685/87faa8e8-9d08-462e-87eb-4dcd384271d8)

constanca-m replied:
Thanks @gizas, I must have overwritten this file by accident because the fields removed are also not matching the description. I checked once again and fixed it

gizas commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -89,56 +44,7 @@
-      description: >
-        Kubernetes Service selectors map
-
-    - name: replicaset.name
```
We need to doublecheck what happens with replicaset for containers when add_resource_metadata.deployment: true, if the container belongs to pod that belongs to replicaset

Or is it only

constanca-m replied:
The field exists

constanca-m commented:
> > Warning: kubernetes.deployment.name had no values, even though there were deployments in the cluster... Maybe there is a bug in the code, as it does not seem to be expected. I did not remove this field. Same for kubernetes.cronjob.name.
> 
> Since elastic-agent-autodiscovery 0.6.4 ([elastic/elastic-agent-autodiscover@285f0bb](https://github.com/elastic/elastic-agent-autodiscover/commit/285f0bb651192115743c7764d60e16e5fe4a0ef9)) we disable deployment and cronjob enrichemnt by default.
> 
> So you need to enable those specifically in the advance options to make the deployment and cronjob name appear.
> 
> > but not for node.annotations.* (same problem as kubernetes.annotations.*).
> 
> The annotations are not include by default. See documentation here: https://www.elastic.co/guide/en/fleet/current/add_kubernetes_metadata-processor.html
> 
> So you need to add include_annotations under node and namespace fields
> 
> FYI there is also this [bug](https://github.com/elastic/elastic-agent/issues/3636) open that is related

Even with EA managed I need to add those by myself? @gizas

constanca-m commented:
> FYI there is also this https://github.com/elastic/elastic-agent/issues/3636 open that is related

I see this is for node labels? I can see them in Discover. Only node annotations are not there

MichaelKatsoulis commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -13,37 +13,11 @@
 - name: kubernetes
   type: group
   fields:
-    - name: pod.name
```
Why did you delete these fields?

constanca-m replied:
It was a mistake, I fixed it now. I think I overwrite this file somehow before making the previous commit.

ChrsMark commented on packages/kubernetes/data_stream/state_pod/fields/base-fields.yml:
```diff
@@ -65,13 +65,6 @@
       description: >
         Kubernetes node labels map
 
-    - name: node.annotations.*
```
I think this will be present if one enables the `add_resource_metadata` setting for the metricset.

constanca-m replied:
I retested it, and you are correct: 
![image](https://github.com/elastic/integrations/assets/113898685/d4ff5296-829a-40ea-8938-365b53f1581d)

I will add this field again.

ChrsMark commented on packages/kubernetes/data_stream/state_node/fields/base-fields.yml:
```diff
@@ -13,37 +13,13 @@
       description: >
         Kubernetes node name
 
-    - name: node.hostname
```
Should this really be remove? Do we know why it is here right now?

constanca-m replied:
I can see if I can find some background on this field. But even checking with `kubectl describe node ...`, there is nothing close to node.hostname

gizas replied:
I would say keep it for now as it might break old versions. If we are not 100% sure

constanca-m replied:
This field was introduced in this [PR](https://github.com/elastic/integrations/pull/569) that modified a lot of files. I had a look at the beats repository and cannot find it there either. I don't know where this field is coming from, but I cannot see any data on it...

ChrsMark reviewed (changes requested):
@constanca-m I would suggest moving the `Changes` part of the PR's description into a more obvious place (in the begging of the description). 

I have left some comments for some fields that should not be removed most probably. We should check the code and ensure that we don't miss sth here. Just testing that the integration works with default settings is not enough for removing stuff.

Also since I see you have a section `Bugs and warnings` with some open questions/concerns why not to first open an issue and call for feedback/discussion there instead of directly opening a PR with "risky" changes? In this way a possible change is first discussed carefully before moving into the actual implementation.

constanca-m commented:
/test

constanca-m commented:
> Just testing that the integration works with default settings is not enough for removing stuff.

I tested all of them again @ChrsMark with EA standalone and I added in the description the manifest I used. In summary, I just added `add_metadata` and `add_resource_metadata` to all data streams. I tested all of them again, and I could find values for `namespace_annotations` for most, but again, no values for `kubernetes.annotations.*` at all.

gizas commented on packages/kubernetes/data_stream/state_container/fields/base-fields.yml:
```diff
@@ -137,8 +130,3 @@
       description: >
         Kubernetes container name
-
-    - name: container.image
```
This is kubernetes.container.name , I agree can be deleted

Not to confuse wtih container.image.name of agent.yml

gizas commented on packages/kubernetes/data_stream/state_daemonset/fields/base-fields.yml:
```diff
@@ -69,35 +45,3 @@
       description: >
         Kubernetes annotations map
-
-    - name: selectors.*
```
Selectors should stay here

According to. https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ daemonset can have selectors

gizas commented on packages/kubernetes/data_stream/state_deployment/fields/base-fields.yml:
```diff
@@ -70,35 +46,8 @@
       description: >
         Kubernetes annotations map
 
-    - name: selectors.*
```
Same for deployment selectors should stay

constanca-m replied:
Here is the [comment](https://github.com/elastic/integrations/pull/8393#discussion_r1381830665) with the link to the implementation PR that introduces `selectors.*` only for `state_service`

gizas replied:
ok

gizas commented on packages/kubernetes/data_stream/state_job/fields/base-fields.yml:
```diff
@@ -70,28 +46,6 @@
       description: >
         Kubernetes annotations map
 
-    - name: selectors.*
```
Same should stay

constanca-m replied:
Here is the [comment](https://github.com/elastic/integrations/pull/8393#discussion_r1381830665) with the link to the implementation PR that introduces `selectors.*` only for `state_service`

gizas commented on packages/kubernetes/data_stream/state_node/fields/base-fields.yml:
```diff
@@ -56,36 +32,4 @@
-      type: keyword
-      description: >-
-        Kubernetes container image
+        Kubernetes annotations map
```
This line should be here?

constanca-m replied:
Yes, it is the description for the `annotations.*` above, the other lines were removed

gizas commented on packages/kubernetes/data_stream/state_persistentvolume/fields/base-fields.yml:
```diff
@@ -55,36 +26,4 @@
-      type: keyword
-      description: >-
-        Kubernetes container image
+        Kubernetes annotations map
```
??

constanca-m replied:
Same as above, the file looks like [this](https://github.com/elastic/integrations/blob/474d24605ff74cab686b5efed575ce9205fa2e14/packages/kubernetes/data_stream/state_persistentvolume/fields/base-fields.yml). It appears confusing with the red and green

gizas commented on packages/kubernetes/data_stream/state_replicaset/fields/base-fields.yml:
```diff
@@ -70,13 +46,6 @@
       description: >
         Kubernetes annotations map
 
-    - name: selectors.*
```
Shoud stay I think

gizas commented on packages/kubernetes/data_stream/state_service/fields/base-fields.yml:
```diff
@@ -13,40 +13,12 @@
       description: >
         Kubernetes namespace
 
-    - name: labels.*
```
Labels and annotations should stay for state_service is not it?

ChrsMark reviewed (commented):
Anything missing from this one? 
If the previous comments are covered maybe resolve them so as to have a clear view of what is pending and what is not.

constanca-m commented:
> Anything missing from this one?

I believe nothing is missing. @ChrsMark 

@gizas Do you want to add anything? I did not mark some of your comments as resolved since I don't know if there is something else you want to say

gizas reviewed (approved):
LGTM, I also did some tests with kind locally and all seem ok
